protocols/messages.go

```diff
@@ -50,8 +50,32 @@ type Message struct {
 	RejectedObjectives []ObjectiveId
 }
 
-// SortedProposals sorts the proposals by channelId and then by turn number.
-func (m Message) SortedProposals() []consensus_channel.SignedProposal {
+// ProposalsSortedForProcessing sorts proposals for a given channel by turn number.
+// It does not swap proposals that target different channels.
+func (m Message) ProposalsSortedForProcessing() []consensus_channel.SignedProposal {
+	proposals := make([]consensus_channel.SignedProposal, len(m.LedgerProposals))
+	copy(proposals, m.LedgerProposals)
+
+	sort.Slice(proposals, func(i, j int) bool {
+		cId1, turnNum1 := proposals[i].ChannelID(), proposals[i].TurnNum
+		cId2, turnNum2 := proposals[j].ChannelID(), proposals[j].TurnNum
+
+		cIdCompare := bytes.Compare(cId1.Bytes(), cId2.Bytes())
+
+		if cIdCompare == 0 {
+			// if the channel ids are the same, sort by turn number
+			return turnNum1 < turnNum2
+		} else {
+			// if the channel ids are different, do not sort
+			return i < j
+		}
+	})
+
+	return proposals
+}
+
+// proposalsSortedForSummary sorts the proposals by channelId and then by turn number.
+func (m Message) proposalsSortedForSummary() []consensus_channel.SignedProposal {
```

Comment:
I'm conflicted about this. On the one hand, it helps us understand that sorting by channel id is only really used for logging purposes and is not used in the core business logic. On the other hand, it means that we log something different to the core business logic, which seems a bit dangerous?

Reply:
A good point. Highly legible misleading information is probably not great.

That said, with updated comments on #1015, I think it's safe to close this alternative off altogether: the problem in 1015 that this PR addressed was a figment of my imagination.